### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by rbuehlma**
@ksooo does this cure your eyes somewhat :)

Before I can merge this, I will need to fix the build for windows/osx. Hopefully @Rechi can help there.

### CODE REVIEW

**Comment by rbuehlma** [comment]
> see [cleanup...Rechi:cleanup](https://github.com/rbuehlma/pvr.zattoo/compare/cleanup...Rechi:cleanup)

Thank you very much :) Seems I was not even close :( Why do we want to keep the travis file now that Jenkins is doing the build?

**Comment by Rechi** [comment]
1. And this is the important one, if you really want to remove Travis CI do it in a separate commit. Then I would have seen yes this wasn't committed by accident.
2. Jenkins doesn't compile build for Linux, only generates the files for our Ubuntu PPA.
3. If you want to disable Travis CI, just do that in there web-interface. That way you can enable later without any code changes.

Please squash the commits properly. I don't want to see my fixup commit (and your fix build commit) in the git history.

**Comment by ksooo** [comment]
> @ksooo does this cure your eyes somewhat :)

 Yep, that could be a good starting point. Thanks.

 How about some more quick wins:
 - never use `using namespace` in header files
 - always use initializer list in ctors
 - remove `this->` from code
 - for parameters, always use references if possible, not values (example `std::string val` vs. `const std::string& val`) => done wrong for example in src/Utils.h
- use `virtual` only where it actually makes sense (if there are no derived classes, remove all `virtual` from methid declarations => done wrong for example in src/Curl.h

**Comment by rbuehlma** [comment]
@Rechi travis fails with the current file. For Linux, I could probably add an "apt install libtinyxml2-dev" but I'm not sure if this would be correct?
For OSX I'm not sure whether there is a similar solution?

File: .travis.yml [hunk 1]
```diff
@@ -29,6 +29,9 @@ matrix:
 # we'll put the Kodi source on the same level
 #
 before_script:
+  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; apt-get update; apt-get install libtinyxml2-dev; fi
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew update; fi
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew install tinyxml2; fi
```
Comment (Rechi) [inline .travis.yml]: This is wrong, will give you the correct changes later.

Comment (rbuehlma) [reply .travis.yml]: Well, I'm curious how it should be :)

Comment (Rechi) [reply .travis.yml]: see https://github.com/rbuehlma/pvr.zattoo/compare/cleanup...Rechi:cleanup again
I overlooked that `CMAKE_DEBUG_POSTFIX` is set for all platforms. Usually libs only set that for windows.

**Comment by rbuehlma** [comment]
@Rechi and @ksooo i now tested this version in runtime. Is it from your point of view ok to merge it?

**Comment by Rechi** [comment]
**NO**, as I already said, squash the commits properly.
### **I don't want to see my fixup commits in the git history.**

**Pull request merged**